test/abstractarray.jl

```diff
@@ -999,3 +999,15 @@ end
 @testset "vcat with mixed elements" begin
     @test vcat(Nothing[], [missing], [1.0], [Int8(1)]) isa Vector{Union{Missing, Nothing, Float64}}
 end
+
+@testset "first/last n elements of vector" begin
```

Comment:
Could you also add tests for non-1-based indices in test/offsetarray.jl?

Reply:
Done!